lib/std/special/compiler_rt/README.md

```diff
@@ -1,15 +1,258 @@
+- dev HackersDelight __addodi4     // a * b, overflow=>ov.*=1 else 0
+- dev HackersDelight __addoti4     // a * b, overflow=>ov.*=1 else 0
+- dev HackersDelight __subosi4     // a * b, overflow=>ov.*=1 else 0
+- dev HackersDelight __subodi4     // a * b, overflow=>ov.*=1 else 0
+- dev HackersDelight __suboti4     // a * b, overflow=>ov.*=1 else 0
+- dev HackersDelight __mulosi4     // a * b, overflow=>ov.*=1 else 0 (required by llvm)
+- dev HackersDelight __mulodi4     // a * b, overflow=>ov.*=1 else 0 (required by llvm)
+- dev HackersDelight __muloti4     // a * b, overflow=>ov.*=1 else 0 (required by llvm)
+
+## Soft float library routines
+
+#### Conversion
+
+- __extendsfdf2  // extend a f32 => f64
+- __extendsftf2  // extend a f32 => f128
+- __extendsfxf2  // extend a f32 => f80 missing
+- __extenddftf2  // extend a f64 => f128
+- __extenddfxf2  // extend a f64 => f80
+- __truncxfdf2   // truncate a to narrower mode of return type, rounding towards zero
+- __trunctfdf2   //
+- __truncxfsf2   // missing
+- __trunctfsf2   //
+- __truncdfsf2   //
+- __fixsfsi      // convert a to i32, rounding towards zero
+- __fixdfsi      //
+- __fixtfsi      //
+- __fixxfsi      // missing
+- __fixsfdi      // convert a to i64, rounding towards zero
+- __fixdfdi      //
+- __fixtfdi      //
+- __fixxfdi      // missing
+- __fixsfti      // convert a to i128, rounding towards zero
+- __fixdfti      //
+- __fixtfdi      //
+- __fixxfti      // missing
+
+- __fixunssfsi   // convert to u32, rounding towards zero. negative values become 0.
+- __fixunsdfsi   //
+- __fixunstfsi   //
+- __fixunsxfsi   // missing
+- __fixunssfdi   // convert to u64, rounding towards zero. negative values become 0.
+- __fixunsdfdi   //
+- __fixunstfdi   //
+- __fixunsxfdi   // missing
+- __fixunssfti   // convert to u128, rounding towards zero. negative values become 0.
+- __fixunsdfti   //
+- __fixunstfdi   //
+- __fixunsxfti   // missing
+
+- __floatsisf    // convert i32 to floating point
+- __floatsidf    //
+- __floatsitf    // missing
+- __floatsixf    //
+- __floatdisf    // convert i64 to floating point
+- __floatdidf    //
+- __floatditf    //
+- __floatdixf    // missing
+- __floattisf    // convert i128 to floating point
+- __floattidf    //
+- __floattixf    //
+
+- __floatunsisf  // convert i32 to floating point
+- __floatunsidf  //
+- __floatunsitf  //
+- __floatunsixf  // missing
+- __floatundisf  // convert i64 to floating point
+- __floatundidf  //
+- __floatunditf  //
+- __floatundixf  // missing
+- __floatuntisf  // convert i128 to floating point
+- __floatuntidf  //
+- __floatuntitf  //
+- __floatuntixf  // missing
+
+#### Comparison
+- __cmpsf2       // missing return (a<b)=>-1,(a==b)=>0,(a>b)=>1,Nan=>1 dont rely on this
+- __cmpdf2       // missing
+- __cmptf2       // missing
+- __unordsf2     // missing (input==NaN) => out!=0 else out=0,
+- __unorddf2     // missing __only reliable for (input!=Nan)__
+- __unordtf2     // missing
```

Comment:
All of these exist?

Reply:
They are then misnamed. For example grepping for `__cmpsf2` I found
```txt
[user@pc compiler_rt]$ rg __cmpsf2
comparesf2_test.zig
23:fn test__cmpsf2(vector: TestVector) bool {
99:        try std.testing.expect(test__cmpsf2(vector));

README.md
195:- __cmpsf2       // missing return (a<b)=>-1,(a==b)=>0,(a>b)=>1,Nan=>1 dont rely on this
```

Reply:
The first three are just exports of a compatible function under a different name:
https://github.com/ziglang/zig/blob/cf5009f9af118716cbe97a28957cb1ca8d047f44/lib/std/special/compiler_rt.zig#L98-L100

The others are handled like any other functions:
https://github.com/ziglang/zig/blob/cf5009f9af118716cbe97a28957cb1ca8d047f44/lib/std/special/compiler_rt.zig#L239-L244

Reply:
ah, good catch. I added that.